client/webserver/webserver.go

```diff
@@ -308,6 +316,56 @@ func New(core clientCore, addr, customSiteDir string, logger dex.Logger, reloadH
 	return s, nil
 }
 
+func (s *WebServer) buildTemplates(lang string) error {
+	langs := make([]language.Tag, 0, 1)
+	dirs := make([]string, 0, 1)
+	var match string
+
+	htmlDir := filepath.Join(s.siteDir, "src", "localized_html")
+
+	fileInfos, err := ioutil.ReadDir(htmlDir)
+	if err != nil {
+		return fmt.Errorf("ReadDir error: %w", err)
+	}
+
+	for _, fi := range fileInfos {
+		if !fi.IsDir() {
+			continue
+		}
+		if fi.Name() == lang {
+			match = fi.Name()
+			break
+		}
+
+		tag, err := language.Parse(fi.Name())
+		if err != nil {
+			log.Debugf("error parsing language tag %q: %v", fi.Name(), err)
+			continue
+		}
+		langs = append(langs, tag)
+		dirs = append(dirs, fi.Name())
+	}
+
+	if match == "" {
+		matcher := language.NewMatcher(langs)
+		_, idx := language.MatchStrings(matcher, lang)
```

Comment:
`MatchStrings` is a bit confusing but it says if no match then "default language is returned if no other language matched.", which is odd (how do we know we didn't pick the default?).

Maybe we should have this be an error, and print the `langs` slice in the event of no match.  This would require cannibalizing `language.MatchStrings` like:

```go
if match == "" {
	// Try to identify candate languages.
	acceptLangs, _, err := language.ParseAcceptLanguage(lang)
	if err != nil {
		return fmt.Errorf("unable to parse requested language: %v", err)
	}
	// Match against template languages.
	matcher := language.NewMatcher(langs)
	tag, idx, conf := matcher.Match(acceptLangs...)
	switch conf {
	case language.Exact:
	case language.High, language.Low:
		log.Infof("Using language %v", tag)
	case language.No:
		return fmt.Errorf("no match for %q in recognized languages %v", lang, langs)
	}
	match = dirs[idx]
}
``` 

Which is just as well because we're only testing for one lang.